crates/common/sync/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use ream_rpc::types::response::BeaconVersionedResponse;
+use ream_storage::{db::ReamDB, tables::Table};
+use tracing::{info, warn};
+pub mod checkpoint;
+
+/// Entry point for checkpoint sync.
+pub async fn initialize_db_from_checkpoint(
+    db: ReamDB,
+    rpc: Option<String>,
+) -> anyhow::Result<(Store, u64)> {
+    let checkpoint_sync_url = match rpc {
+        Some(url) => url,
+        // 0 for Mainnet, 1 for Sepolia
+        None => fetch_default_checkpoint_url().expect("Unable to fetch default checkpoint url")[1]
+            .checkpoint_urls[0]
+            .clone(),
+    };
+
+    info!("Starting Checkpoint Sync");
+    let current_slot = db
+        .slot_index_provider()
+        .get_highest_slot()
+        .map_err(|err| anyhow!("Unable to fetch highest slot in db:{}", err))?;
+
+    if current_slot.is_some() {
+        warn!("Starting Checkpoint Sync from existing DB. It is advised to start from a fresh DB");
+    }
+    let block = fetch_finalized_block(&checkpoint_sync_url, current_slot).await?;
+    let slot = block.data.message.slot;
+    fetch_blobs(
+        &checkpoint_sync_url,
+        db.clone(),
+        block.data.message.block_hash(),
+    )
+    .await?;
+
+    let state = fetch_finalized_state(&checkpoint_sync_url, block.data.message.slot).await?;
+    info!("Received block from slot:{}", slot);
+    ensure!(
+        block.data.message.slot == state.data.slot,
+        anyhow::anyhow!("Slot mismatch")
+    );
+
+    ensure!(block.data.message.state_root == state.data.state_hash());
+    let mut store = get_forkchoice_store(state.data, block.data.message, db)?;
+
+    // using sepolia genesis time until pectra mainnet
+    let time = network_spec().genesis.genesis_time + SECONDS_PER_SLOT * (slot + 1);
+    on_tick(&mut store, time)?;
+    info!("Initial Sync complete");
+
+    Ok((store, slot))
+}
+
+/// Fetch initial state from trusted RPC
+pub async fn fetch_finalized_state(
+    rpc: &str,
+    slot: u64,
+) -> anyhow::Result<BeaconVersionedResponse<BeaconState>> {
+    let body = reqwest::get(format!("{}{}{}", rpc, "/eth/v2/debug/beacon/states/", slot))
+        .await?
+        .text()
+        .await?;
+
+    let state: BeaconVersionedResponse<BeaconState> = serde_json::from_str(&body)?;
+    info!("State hash:{}", state.data.state_hash());
+
+    Ok(state)
+}
+
+/// Fetch initial block from trusted RPC
+pub async fn fetch_finalized_block(
+    rpc: &str,
+    slot: Option<u64>,
+) -> anyhow::Result<BeaconVersionedResponse<SignedBeaconBlock>> {
+    let url = match slot {
+        Some(slot) => {
+            info!("Starting from slot:{}", slot);
+            format!("{}{}{}", rpc, "/eth/v2/beacon/blocks/", slot + 1)
+        }
+        None => {
+            format!("{}{}", rpc, "/eth/v2/beacon/blocks/finalized")
+        }
+    };
+
+    let body = reqwest::get(url).await?.text().await?;
+
+    let mut block: BeaconVersionedResponse<SignedBeaconBlock> = serde_json::from_str(&body)?;
```

Comment:
```suggestion
pub async fn fetch_finalized_block(
    rpc: &str,
) -> anyhow::Result<BeaconVersionedResponse<SignedBeaconBlock>> {
    let block = reqwest::get(format!("{rpc}/eth/v2/beacon/blocks/finalized")).await?.json<BeaconVersionedResponse<SignedBeaconBlock>>().await?;
```

are we fetching a finalized block or any block? the function name says finalized